trieste/models/gpflow/interface.py

```diff
@@ -71,9 +71,15 @@ def create_posterior_cache(self) -> None:
         """
         self._posterior = self.model.posterior(PrecomputeCacheType.VARIABLE)
 
+    @abstractmethod
+    def _ensure_variable_model_data(self) -> None:
+        """Ensure GPflow data, which is normally stored in Tensors, is instead stored in
+        dynamically shaped Variables."""
+
     def __setstate__(self, state: dict[str, Any]) -> None:
         # when unpickling we may need to regenerate the posterior cache
         self.__dict__.update(state)
+        self._ensure_variable_model_data()
```

Comment:
Question: how do we ensure that after the deepcopy, the posterior cache will have dynamic shape, or is that not required?

Reply:
Never mind.. I didn't process the following 2 lines correctly. It is clear now how that works :-)